tools/packaging/kata-deploy/binary/src/utils/toml.rs

```diff
@@ -0,0 +1,1605 @@
+// Copyright (c) 2019 Kata Containers community
+// Copyright (c) 2025 NVIDIA Corporation
+//
+// SPDX-License-Identifier: Apache-2.0
+
+use anyhow::{Context, Result};
+use std::path::Path;
+use toml_edit::{DocumentMut, Item, Value};
+
+/// Parse a TOML path into components, respecting quoted keys
+/// Example: `.plugins."io.containerd.cri.v1.runtime".containerd.runtimes."kata"`
+/// Results in: ["plugins", "io.containerd.cri.v1.runtime", "containerd", "runtimes", "kata"]
+fn parse_toml_path(path: &str) -> Result<Vec<String>> {
+    let path = path.trim_start_matches('.');
+    let mut parts = Vec::new();
+    let mut current = String::new();
+    let mut in_quotes = false;
+    let mut chars = path.chars().peekable();
+
+    while let Some(ch) = chars.next() {
+        match ch {
+            '"' => {
+                in_quotes = !in_quotes;
+                // Don't include the quote characters themselves
+            }
+            '.' if !in_quotes => {
+                if !current.is_empty() {
+                    parts.push(current);
+                    current = String::new();
+                }
+            }
+            _ => {
+                current.push(ch);
+            }
+        }
+    }
+
+    if !current.is_empty() {
+        parts.push(current);
+    }
+
+    if in_quotes {
+        return Err(anyhow::anyhow!("Unmatched quotes in TOML path: {path}"));
+    }
+
+    if parts.is_empty() {
+        return Err(anyhow::anyhow!("Invalid TOML path: {path}"));
+    }
+
+    Ok(parts)
+}
+
+/// Set a TOML value at a given path (e.g., ".plugins.cri.containerd.runtimes.kata.runtime_type")
+pub fn set_toml_value(file_path: &Path, path: &str, value: &str) -> Result<()> {
```

Comment:
With this approach, you re-parse the TOML document every time, and write it to disk every single time.

Wouldn't it make sense to parse the document once, do the various adjustments (possibly using `toml_edit` built-in indexing) and then write the whole thing at the end?

```
use toml_edit::{DocumentMut, value};

let mut doc = content.parse::<DocumentMut>()?;

// pass doc around, and do stuff like
doc["plugins"]["io.containerd.cri.v1.runtime"]["containerd"]["runtimes"]["kata"]["runtime_type"] = value("io.containerd.kata.v2");
```

Reply:
@c3d, may I bother you to open an issue for this? This is something I'd like to address as a follow-up improvement, and it does make sense to be improved.